bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -446,13 +446,13 @@ public void run() {
             executor.shutdown();
             try {
                 if (!executor.awaitTermination(environment.getTimeout(), TimeUnit.SECONDS)) {
-                    // For some reason we've timed out. The deletion should likely be executing, but let's force it to
-                    // be safe.
-                    cleaner.cleanup();
+                    // For some reason we've timed out. The deletion should likely be executing.
+                    // We can't start a new cleanup to force it. On Windows we would have the side effect to have 2 cleaner processes to
+                    // be executed, with the risk that a new installation has been installed and the new cleaner cleaning the new installation
+                    log.cleanupTimeout(environment.getTimeout(), environment.getJBossHome());
                 }
-            } catch (IOException | InterruptedException e) {
-                // Possibly already logged, but we should log again to be safe
-                log.failedToStartCleanupProcess(e, environment.getJBossHome());
+            } catch (InterruptedException e) {
+                // The task has veen interrupted, leaving
```

Comment:
I do think we should leave the log message here too, but the above change makes sense as well.

Reply:
Good! I added it back.